### PR TITLE
added roles, access by role, role entity

### PR DESCRIPTION
**Pull request by DmytroHadiuchko**

### CODE REVIEW

File: src/main/java/dmytro/hadiuchko/springboot/entity/Role.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.springframework.security.core.GrantedAuthority;
+
+@Entity
+@Data
```
Comment (TonyH277) [inline src/main/java/dmytro/hadiuchko/springboot/entity/Role.java]:
```suggestion
@Entity
@Getter
@Setter
```

File: src/main/java/dmytro/hadiuchko/springboot/entity/User.java [hunk 2]
```diff
@@ -2,16 +2,28 @@
...
+import java.util.Collection;
+import java.util.Set;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
 
 @Data
 @Entity
```
Comment (TonyH277) [inline src/main/java/dmytro/hadiuchko/springboot/entity/User.java]: the same

File: src/main/java/dmytro/hadiuchko/springboot/entity/User.java [hunk 3]
```diff
@@ -25,4 +37,48 @@ public class User {
...
+
+    @Override
+    public String getUsername() {
+        return email;
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
```
Comment (TonyH277) [inline src/main/java/dmytro/hadiuchko/springboot/entity/User.java:65]: I am not sure but Data annotation will create getter for Passord field, I think if we haven't got password field we should override getPassword method and return credentials we are using for auth

**Review by TonyH277 — APPROVED** [review]
LGFM!

File: src/main/java/dmytro/hadiuchko/springboot/entity/Role.java (hunk 1 above)
Comment (andrii-hoienko) [reply src/main/java/dmytro/hadiuchko/springboot/entity/Role.java]: Not resolved

File: src/main/java/dmytro/hadiuchko/springboot/entity/Role.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @Column(nullable = false, unique = true)
+    @Enumerated(EnumType.STRING)
+    private RoleName name;
+
+    @Override
+    public String getAuthority() {
+        return name.name();
+    }
+}
+
```
Comment (andrii-hoienko) [inline src/main/java/dmytro/hadiuchko/springboot/entity/Role.java]:
```suggestion

```

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**